src/main/plugins/org.dita.html5/xsl/topic.xsl

```diff
@@ -2358,12 +2358,12 @@ See the accompanying LICENSE file for applicable license.
       <xsl:call-template name="generateDefaultMeta"/> <!-- Standard meta for security, robots, etc -->
       <xsl:apply-templates select="." mode="getMeta"/> <!-- Process metadata from topic prolog -->
       <xsl:call-template name="copyright"/>         <!-- Generate copyright, if specified manually -->
-      <xsl:call-template name="generateCssLinks"/>  <!-- Generate links to CSS files -->
       <xsl:call-template name="generateChapterTitle"/> <!-- Generate the <title> element -->
       <xsl:call-template name="gen-user-head" />    <!-- include user's XSL HEAD processing here -->
       <xsl:call-template name="gen-user-scripts" /> <!-- include user's XSL javascripts here -->
       <xsl:call-template name="gen-user-styles" />  <!-- include user's XSL style element and content here -->
       <xsl:call-template name="processHDF"/>        <!-- Add user HDF file, if specified -->
+      <xsl:call-template name="generateCssLinks"/>  <!-- Generate links to CSS files -->
```

Comment:
Should we instead move `processHDF` up instead of moving `generateCssLinks` down. Then this would allow you to declared _everything_ before the normal `gen*` template output. So move `generateHDF` to after `generateCharset`?

Reply:
If we consider **only** `processHDF` and `generateCssLinks`, then yes: that would solve the problem just as well — but if anyone were to embed an external stylesheet using `gen-user-head` or `gen-user-styles` the same problem would appear — the custom CSS file would not override the framework stylesheets, since they'd be listed _later_.

Though the template names _suggest_ certain usage, any one of them could theoretically be used to insert arbitrary code into the `<head>` element, including linked stylesheets.

Since the `gen-user-*` templates are all empty by default, it's hard to say whether these should precede or follow  `processHDF`, but I'd be inclined to preserve the original order there in case there's a reason for that sequence.

Apart from charset first, I'm not sure if there's any rule or best practice convention regarding the order of elements within an HTML `<head>`, but I think it's fairly common for the `<title>` element to precede linked stylesheets.

Reply:
I can't say for sure, but my guess is that there is not a reason for the original sequence with those gen-* stubs; I think it's likely historical and just hasn't changed. I think it's probably safe to say if you specify a CSS file with your parameter, that's the one you want to have the most weight. I wonder though, if you've decided to go with `gen-user-styles` to put in one or more CSS links, those would previously override `commonltr.css` but after this move, the common one would win, right?

As for order within `<head>`, the title is supposed to come after the character set metadata so that any tool reading it already knows the character set. Otherwise I haven't run into rules or best practices for the order of these.